src/CSET/_workflow_utils/finish_website.py

```diff
@@ -2,14 +2,58 @@
 
 """Write finished status to website front page.
 
-Does the final update to the workflow status on the front page of the web
-interface.
+Constructs the plot index, and does the final update to the workflow status on
+the front page of the web interface.
 """
 
+import json
+import logging
 import os
+from pathlib import Path
 
+from CSET._common import combine_dicts
 
-def run():
-    """Run workflow script."""
-    with open(f"{os.getenv('WEB_DIR')}/status.html", "wt") as fp:
+logging.basicConfig(
+    level=os.getenv("LOGLEVEL", "INFO"), format="%(asctime)s %(levelname)s %(message)s"
+)
+
+
+def construct_index():
+    """Construct the plot index.
+
+    Index should has the form {"Category Name": {"recipe_id": "Plot Name"}}
+    """
+    index = {}
+    plots_dir = Path(os.environ["CYLC_WORKFLOW_SHARE_DIR"] + "/web/plots")
+    # Loop over all directories, and append to index.
+    for directory in (d for d in plots_dir.iterdir() if d.is_dir()):
```

Comment:
Don't think iterdir is recursive, will there be a future case where we use more complex subdirectories to store plots (suspect not)

Reply:
Iterdir is not recursive. I don't want to go too deep as each plot could (theoretically) have as many folders within as it wants.

Reply:
```suggestion
    # Only visits the directories directly under the plots directory.
    for directory in (d for d in plots_dir.iterdir() if d.is_dir()):
```